src/components/infinite-scroll/InfiniteScroll.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * From https://github.com/jaredpalmer/react-simple-infinite-scroll
+ * Updated to accept a scroll container React ref as the "window"
+ */
+import * as React from 'react';
+import throttleFn from 'lodash/throttle';
+
+export interface InfiniteScrollProps {
+    /** Components to render, should include entities/loading states/end message */
+    children: React.ReactNode;
+    /** Does the resource have more entities */
+    hasMore: boolean;
+    /** Are more entities being loaded */
+    isLoading: boolean;
+    /** Callback to load more entities */
+    onLoadMore: () => void;
+    /** React ref of the scroll container.
+     * Used to listen to scroll events and calculate onLoadMore calls.
+     * Set useWindow to true to use the window as scroll container.
+     */
+    scrollContainerNode?: HTMLElement;
+    /** Scroll threshold */
+    threshold?: number;
+    /** Throttle rate */
+    throttle?: number;
+    /** Set useWindow to true to use the window as scroll container. If set to true, will ignore scrollContainerRef. */
+    useWindow?: boolean;
+}
+
+type OnContainerScrollParams = Pick<Required<InfiniteScrollProps>, 'threshold' | 'useWindow'> &
+    Pick<InfiniteScrollProps, 'isLoading' | 'hasMore' | 'scrollContainerNode' | 'onLoadMore'> & {
+        sentinelRef: React.RefObject<HTMLDivElement>;
+    };
+
+function onContainerScroll({
```

Comment:
Would it make sense to still keep this inside the `InfiniteScroll` component but before the `React.useEffect` block? I.e. declare this under `const sentinelRef` on line 74.

This way we don't need to pass in any params since we should have access to the `InfiniteScroll` props.

Reply:
@tjuanitas , I suggested extracting it outside of the component: 
https://github.com/box/box-ui-elements/pull/2881#discussion_r861718607 
What d'you think?

Reply:
Gotcha. Yeah, I thought it might look cleaner since we're always passing in the same parameters from the parent props when we call `onContainerScroll`. The calls when we declare `scrollHandler` and `resizeHandler` and any new ones could then be condensed.

Also, this is preference, but the declaration for `OnContainerScrollParams` looked odd to me lol.

I'm okay with leaving it outside though if there's a benefit to it.

Reply:
It would be nice not to have such chunky `scrollHandler` and `resizeHandler` calls. Could we condense those props (except for sentinelRef) using a rest parameter? Here is the case that I made for breaking the function out:
> `onContainerScroll` has closure from `InfiniteScroll`. This makes it possible to accidentally make use of sideEffects from InfiniteScroll. Moving it out of the closure and requiring its dependencies to be passed as parameters protects against this hazard. It also has the added benefits of making that function testable and reusable.

We may not end up unit testing or reusing `onContainerScroll`; so I think, for the most part, the question here is, how valuable is it to isolate functions from the side effects they would have access to if composed in the scope of the component?

Reply:
I personally like the picks and intersections that @apieceofbart used in that declaration. I also can't think of an alternative except for duplicating the types 🤔

Reply:
Hey @apieceofbart, I talked with @jcamden and let's keep this as-is but can we sort the parameters? Here and in the declarations for `scrollHandler` and `resizeHandler`.